### PR TITLE
Order framework events by ID in jobmanager status/resume methods

### PR DESCRIPTION
**Pull request by rihter007**
Because ID is a new field, we are breaking contracts (a bit)

### CODE REVIEW

File: pkg/jobmanager/status.go [hunk 1]
```diff
@@ -84,22 +84,30 @@ func (jm *JobManager) status(ev *api.Event) *api.EventResponse {
 		completionEvents[eventName] = true
 	}
 
-	for _, ev := range jobEvents {
+	var lastJobEventIdx int
+	for idx, ev := range jobEvents {
 		if ev.EventName == job.EventJobStarted {
-			startTime = ev.EmitTime
+			if ev.EmitTime.After(startTime) {
```
Comment (tfg13) [inline pkg/jobmanager/status.go:90]: why do we need to check this?

File: pkg/jobmanager/status.go [hunk 2]
```diff
@@ -84,22 +84,30 @@ func (jm *JobManager) status(ev *api.Event) *api.EventResponse {
...
+				startTime = ev.EmitTime
+			}
 		} else if _, ok := completionEvents[ev.EventName]; ok {
 			// A completion event has been seen for this Job. Only one completion event can be associated to the job
-			if endTime != nil && !endTime.IsZero() {
+			if endTime != nil || !endTime.IsZero() {
```
Comment (tfg13) [inline pkg/jobmanager/status.go:95]: won't this panic if `endTime` is `nil`?
https://play.golang.org/p/MZ8S8qPyexp

**Review by tfg13 — CHANGES_REQUESTED** [review]

File: pkg/jobmanager/status.go (hunk 1 above)
Comment (rihter007) [reply pkg/jobmanager/status.go:90]: what if we have multiple events?

Comment (rihter007) [reply pkg/jobmanager/status.go]: Ok, let's keep it as is

Comment (mimir-d) [reply pkg/jobmanager/status.go]: multiple job start events? that should be a bug

**Review by mimir-d — APPROVED** [review]

**Review by tfg13 — APPROVED** [review]

**Pull request merged**